contracts/token/CRC1155/ICRC1155Enumerable.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC1155/IERC1155.sol";
+
+/**
+ * @dev Extension of ERC1155 to allow enumerability.
+ */
+interface ICRC1155Enumerable is IERC1155 {
+    /**
+     * @dev Returns the total amount of tokens stored by the contract.
+     */
+    function totalSupply() external view returns (uint256);
```

Comment:
If the contract has 2 `A` tokens and 3 `B` tokens then is the total supply 2 or 5?

Reply:
`totalSupply()` returns 5, and there is another API `totalSupply(uint256 tokenId)` that return total supply of specified token Id.

Reply:
So `tokenCountOf` gives the upper limit to use with `tokenOfOwnerByIndex `. What about `tokenByIndex`? Shouldn't there be an API like `totalTokenCount`?

I think `totalSupply()` is confusing and I'm not sure it should be part of this extension. I'm also not sure what would be the use case for it.

Also, I find the naming confusing, mostly because "token" can mean both the fungible token and the non-fungible tokenId. Maybe something like this would be better:

`totalTokenIdCount`, `tokenIdByIndex`, `tokenIdCountOf`, `tokenIdOfOwnerByIndex`

Reply:
All the method names come from OZ latest code or discussions, I want to keep the naming consistent with OZ. Once OZ supported the enumerable interfaces, users could easily upgrade to the OZ edition.

Reply:
I'm just concerned that the current naming might lead to some confusion. It's only a weak preference though, so feel free to go with the current version.